File: python DSA/binarySearch.py
# -*- coding: utf-8 -*-
"""
Created on Tue Nov 24 21:59:11 2020

@author: Venkateswarlu
"""

def binarySearch(a,x,n):
    left=0
    right=n-1
    flag=False
    while left<=right :
        if left!=right:
            middle=left+(right-1)//2
        elif left==right:
            middle=left
        if x<a[middle]:
            right=middle-1
        elif x>a[middle]:
            left=middle+1
        elif x==a[middle]:
            print("ele"+str(x)+"is present at index= "+str(middle))
            flag=True
            break
    if flag==False:
        print("NO RECORD OF "+str(x)+"is found in list");
    elif flag==True:
        print("records are found and displayed above")
        
    
  
    
a=[ ]
n=int(input("enter no of ele in list\n"))
for i in range(n):
    print("enter your "+str(i)+"ele of list\n");
    ele=int(input())
    a.append((ele))
i=0
print("\n")
for i in range(n):
    print(a[i])
print("enter ele to be searched for\n")
x=int(input())
binarySearch(a,x,n)
    

File: c DSA/binarySearch.cpp
//BINARY SEARCH(SEARCHING ALGORITHMS)(//ASSUMING INPUT TO BE IN ASCENDING ORDER)
#include<stdio.h>
void print(int a[],int n)
{
	for(int i=0;i<n;i++)
	{
		printf("%d ",a[i]);
	}
}
void binarySearch(int a[],int x,int n)
{
	int left=0,right=n-1,middle;
	bool flag;
	while(left<=right)
	{ if(left!=right){middle=left+(right-1)/2;;
}
else if(left==right){
	middle=left;
}
	if(a[middle]==x) { flag=true;
	printf("\n%d is found at index= %d\n",x,middle);
	break;
	}
	if(x>a[middle]) left=middle+1;
	if(x<a[middle]) right=middle-1;	
	
	}
	if(flag==false) printf("\n%d is not found at all in array \n",x);
	else if(flag==true) printf("\n%d is found and displayed above\n",x);
		
	
}
int main()
{
	int a[10],n,x;
	printf("enter no of ele \n");
	scanf("%d",&n);
	for(int i=0;i<n;i++)
	{ printf("enter ele into array\n");
		scanf("%d",&a[i]);
	}
	printf("enter ele you want to search\n");
	scanf("%d",&x);
	print(a,n);
	binarySearch(a,x,n);
	
}
